server/repository/userRepository.js

```diff
@@ -90,39 +128,50 @@ class UserRepository {
 
   /**
    * This method finds all non-admin users in the database
-   *@returns {object | null} the results from DB
-  */
-  static async getAllUsers() {
-    const allUsers = await User.findAll({
+   * Finds a user using the userId
+   * @param {number} username userId
+   * @returns {object | null} the details of the user.
+   */
+  static async getUserbyUsername(username) {
+    const oneUser = await User.findOne({
       where: {
         role: 'user',
+        username,
       },
-      attributes: {
-        include: [
-          'id',
-          'firstName',
-          'lastName',
-          'facebook',
-          'google',
-          'twitter',
-          'bio',
-          'imageUrl',
-          'createdAt',
-          'updatedAt',
-        ],
-        exclude: [
-          'email',
-          'updatedAt',
-          'deletedAt',
-          'isDeleted',
-          'password',
-          'emailConfirmation',
-          'resetToken',
-          'role',
-        ],
+      attributes: [
+        'id',
+        'username',
+        'firstName',
+        'lastName',
+        'email',
+        'isConfirmed',
+        'facebook',
+        'google',
+        'twitter',
+        'bio',
+        'imageUrl',
+        'createdAt',
+        'updatedAt',
+      ],
+    });
+    if (!oneUser) return null;
+    return oneUser;
+  }
+
+  /**
+   * Updates the user profile
+   * @param { object } newData
+   * @param { string } username
+   * @returns { boolean } true for successful update.
+   */
+  static async updateUser(newData, username) {
```

Comment:
So are we creating a different endpoint to update an admin user?

Reply:
Not really.  We can use this endpoint.
Thanks.